pkg/api/http/http.go

```diff
@@ -1794,30 +1788,35 @@ func (a *api) onBulkPublish(reqCtx *fasthttp.RequestCtx) {
 			}
 			bulkRes.FailedEntries = append(bulkRes.FailedEntries, resEntry)
 		}
-		status := nethttp.StatusInternalServerError
 		bulkRes.ErrorCode = "ERR_PUBSUB_PUBLISH_MESSAGE"
 
-		if errors.As(err, &runtimePubsub.NotAllowedError{}) {
-			msg := NewErrorResponse("ERR_PUBSUB_FORBIDDEN", err.Error())
-			status = nethttp.StatusForbidden
-			fasthttpRespond(reqCtx, fasthttpResponseWithError(status, msg), closeChildSpans)
-			log.Debug(msg)
-
+		switch {
+		case errors.As(err, &runtimePubsub.NotAllowedError{}):
+			nerr := apierrors.PubSubPublishForbidden(pubsubName, string(contribMetadata.PubSubType), topic, a.universal.AppID(), err)
+			standardizedErr, ok := kiterrors.FromError(nerr)
+			if ok {
+				fasthttpRespond(reqCtx, fasthttpResponseWithError(standardizedErr.HTTPStatusCode(), standardizedErr), closeChildSpans)
+			}
+			log.Debug(nerr)
 			return
-		}
-
-		if errors.As(err, &runtimePubsub.NotFoundError{}) {
-			msg := NewErrorResponse("ERR_PUBSUB_NOT_FOUND", err.Error())
-			status = nethttp.StatusBadRequest
-			fasthttpRespond(reqCtx, fasthttpResponseWithError(status, msg), closeChildSpans)
-			log.Debug(msg)
-
+		case errors.As(err, &runtimePubsub.NotFoundError{}):
+			nerr := apierrors.PubSubTestNotFound(pubsubName, string(contribMetadata.PubSubType), topic, err)
+			standardizedErr, ok := kiterrors.FromError(nerr)
+			if ok {
+				fasthttpRespond(reqCtx, fasthttpResponseWithError(standardizedErr.HTTPStatusCode(), standardizedErr), closeChildSpans)
+			}
 			return
+		default:
+			err = apierrors.PubSubPublishMessage(pubsubName, string(contribMetadata.PubSubType), topic, err)
+			log.Debug(err)
 		}
 
 		// Return the error along with the list of failed entries.
 		resData, _ := json.Marshal(bulkRes)
-		fasthttpRespond(reqCtx, fasthttpResponseWithJSON(status, resData, nil), closeChildSpans)
+		standardizedErr, ok := kiterrors.FromError(err)
```

Comment:
Just a nit; maybe this can be more concise as:

```
if standardizedErr, ok := kiterrors.FromError(err); ok {
    fasthttpRespond(reqCtx, fasthttpResponseWithJSON(standardizedErr.HTTPStatusCode(), resData, map[string]string{"responseData": string(resData), "error": standardizedErr.Error()}), closeChildSpans)
}
```